lib/external-housing-available/external-housing-available.controllers.js

```diff
@@ -1,6 +1,172 @@
+      }
+
+      if (process.env.SUPER_ADMIN !== request.auth.credentials.subjectId) {
+        throw Boom.forbidden('Must be an Admin Account to post an External Housing Avaliable');
+      }
+
+      //Match address input with geocoder.
+      var geocodes = await postalAddressLib.addressGeocode(request.payload.address);
+      //If address doesnt exist or length is 0. 
+      if (!geocodes.results || geocodes.results.length === 0) {
+        try {
+          //Log it into the logger
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.no_results',
+            severity: 'Notice',
+            request: request,
+          });
+        } catch (e) {}
+        throw Boom.badData('No results for that address');
+      } 
+
+      const firstGeocode = geocodes.results[0];
+
+      let addressObject = postalAddressLib.extractAddress(firstGeocode);
+
+      if (!addressObject.addressNumber) {
+        try {
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.street_number',
+            severity: 'Notice',
+            request: request,
+          });
+        } catch (e) {}
+        throw Boom.badData('Address is missing a street number');
+      }
+
+      //Address without a county, we want to use locality instead.
+      if (!addressObject.hasOwnProperty('addressCounty')) {
+        addressObject.addressCounty = addressObject.addressLocality;
+      }
+      if (
+        addressObject.addressRegion !== process.env.ADDRESS_LIMIT_STATE ||
+        addressObject.addressCounty !== process.env.ADDRESS_LIMIT_COUNTY
+      ) {
+        try {
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.county',
+            severity: 'Notice',
+            request: request,
+          }); 
+        } catch (e) {} 
+        throw Boom.badData('Sorry, We are only accepting properties in ' + `${process.env.ADDRESS_LIMIT_COUNTY}, ${process.env.ADDRESS_LIMIT_STATE}`); 
+      }
+
+      //Extracting the address using function from geoCoordinates lib
+      let geoCoordObject = geoCoordinatesLib.extractLocation(firstGeocode);
+      
+      const addressHash = postalAddressLib.hashAddress(addressObject);
+      addressObject.hash = addressHash;
+
+      // External Housing Available Object 
+      let externalHousingAvailableObject = {};
+
+      const existingAddresses = await postalAddressLib.getAddressesByHash(addressHash);
+      if (existingAddresses.length > 0) {
+        //If this is an exisiting address
+        externalHousingAvailableObject.PropertyId = existingAddresses[0].PropertyId;
+      } else {
+        //Else, create a new propertyId for non exisiting address
+      }
+
+      const standardizedAddress = postalAddressLib.standardizeAddress(addressObject);
+      addressObject.address = standardizedAddress; 
+
+      //Title
+      if (request.payload.title && request.payload.title !== request.payload.address) {
```

Comment:
I think we just want the title to be set as the title. the standardized address has unit information stripped from it, so we don't really want to use it too much on housing-available (which will presumably have unit info on it).